Sinch bak

nicksenap opened the pull request.

nicksenap committed:
feat(api): add Sinch client for sending RCS messages

Introduce a new Sinch client to send RCS messages using the Sinch
Conversation API. This includes initializing the client with necessary
credentials and implementing a method to send messages.

feat(routes): add endpoint to send SMS using Sinch

Add a new endpoint `/test/sinch` to send SMS messages using the Sinch
client. This allows testing of the Sinch integration via a simple API
call.

chore(config): add Sinch API credentials to polytope.yml

Include necessary environment variables for Sinch API credentials in
the polytope.yml configuration file to support the new Sinch client
functionality.

nicksenap committed:
Merge remote-tracking branch 'origin/main' into sinch_bak

aeriksson commented on polytope.yml:
```diff
@@ -33,6 +33,11 @@ modules:
         - { name: OPENAI_API_KEY, value: "#pt-secret openai-api-key" }
         - { name: ANTHROPIC_API_KEY, value: "#pt-secret anthropic-api-key" }
         - { name: FOOTWAY_API_KEY, value: "#pt-secret footway-api-key" }
+        - { name: SINCH_API_KEY_ID, value: "#pt-secret sinch-api-key-id" }
+        - { name: SINCH_API_KEY_SECRET, value: "#pt-secret sinch-api-key-secret" }
+        - { name: SINCH_API_PROJECT_ID, value: "#pt-secret sinch-api-project-id" }
+        - { name: SINCH_APP_ID, value: "#pt-secret sinch-app-id" }
+        - { name: SINCH_SMS_SENDER, value: "#pt-secret sinch-sms-sender" }
```
I'm a bit reluctant to accept this at this point because this step will refuse to run if these secrets haven't been stored. Could you set these to empty strings instead? That way people who want to work with Sinch can still do so without it breaking things for others

nicksenap committed:
refactor(sinch.py): remove unused sms_sender attribute and from_number parameter to simplify RCS sending logic
docs(sinch.py): update docstring to reflect changes from SMS to RCS and remove from_number argument description
